### PR TITLE
Refactor Stax tests and bump version

### PR DESCRIPTION
**Pull request by ajinkyaraj-23**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -28,9 +28,10 @@ jobs:
       - name: Build
         run: |
           DEVICE_SDK="$(echo ${{ matrix.device }} | tr a-z A-Z)_SDK"
-          bash -c "make -C app BOLOS_SDK=\$${DEVICE_SDK}"
+          DEVICE_TARGET=${{ matrix.device }}
+          bash -c "TARGET=\$${DEVICE_TARGET} make -C app BOLOS_SDK=\$${DEVICE_SDK}"
           tar -C app/bin -czf app_${{ matrix.device }}.tgz .
-          bash -c "make -C app BOLOS_SDK=\$${DEVICE_SDK} DEBUG=1"
+          bash -c "TARGET=\$${DEVICE_TARGET} make -C app BOLOS_SDK=\$${DEVICE_SDK} DEBUG=1"
```
Comment (spalmer25) [inline .github/workflows/build.yml:34]: The `TARGET` is not required here.
The distinction between `Stax` and `Flex` is done by the `BOLOS_SDK`: `STAX_SDK` and `FLEX_SDK`

File: README.md [hunk 2]
```diff
@@ -215,6 +215,9 @@ You will be requested to press enter to take each snapshot in term.
 **NB** make sure that the screen has updated to the screen you want to snapshot each time. It's also a good idea to
 re-run the test normally afterwards, to ensure the snapshots have been set correctly.
 
+
+To reset the home and info screens, delete the home.png and info.png from test/integration/stax/snapshots folder and generate golden images.
+
```
Comment (spalmer25) [inline README.md:220]: Why such behaviour?

File: tests/requirements.txt [hunk 3]
```diff
@@ -3,6 +3,6 @@ setuptools
 pkginfo
 cryptography
 base58
-pytezos==3.10.3
+pytezos
```
[the remark this answers is not in the document]
Comment (ajinkyaraj-23) [reply tests/requirements.txt:6]: Tests are passing with the latest version so no need to fix a version

File: README.md (hunk 2 above)
Comment (ajinkyaraj-23) [reply README.md:220]: Only one copy of each file is desired and thus they are kept in parent folder. which makes deleting them difficult

Comment (spalmer25) [reply README.md:220]: I understand why some snapshots are kept in `stax/snapshots`, but why would anyone delete them manually?

File: .github/workflows/build.yml (hunk 1 above)
Comment (ajinkyaraj-23) [reply .github/workflows/build.yml]: Removed.

File: README.md (hunk 2 above)
Comment (ajinkyaraj-23) [reply README.md]: Added automatic deletion of these files.

**Review by spalmer25 — COMMENTED** [review]
Can you modify/complete the title/description of the PR so that it better corresponds to what it does now?
Apart from a few comments, LGTM

File: tests/integration/nano/utils/app.py [hunk 4]
```diff
@@ -23,14 +23,14 @@
 from requests.exceptions import ConnectionError
 from ragger.backend import SpeculosBackend
 from ragger.error import ExceptionRAPDU
-from ragger.navigator import NavInsID, NanoNavigator
+from ragger.navigator import NavInsID, NanoNavigator, TouchNavigator
```
[the remark this answers is not in the document]
Comment (ajinkyaraj-23) [reply tests/integration/nano/utils/app.py:26]: Yes its unused. I plan to add stax test in the same way as nano tests are run currently. This is setting the background for now.

File: tests/integration/nano/conftest.py [hunk 5]
```diff
@@ -19,7 +21,7 @@ def pytest_addoption(parser):
     parser.addoption("-D", "--device",
                      type=str,
                      choices=DEVICES,
-                     help="Device type: nanos | nanosp | nanox",
+                     help="Device type: nanos | nanosp | nanox | stax | flex",
```
[the remark this answers is not in the document]
Comment (ajinkyaraj-23) [reply tests/integration/nano/conftest.py:24]: Ok.

File: tests/integration/nano/conftest.py [hunk 6]
```diff
@@ -11,6 +11,8 @@
     Firmware.NANOS,
     Firmware.NANOSP,
     Firmware.NANOX,
+    Firmware.STAX,
+    Firmware.FLEX
```
[the remark this answers is not in the document]
Comment (ajinkyaraj-23) [reply tests/integration/nano/conftest.py:15]: Ok.

[the remark this answers is not in the document]
Comment (ajinkyaraj-23) [reply tests/integration/nano/conftest.py]: Removed.

File: tests/integration/nano/conftest.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ajinkyaraj-23) [reply tests/integration/nano/conftest.py]: Removed.

**Review by spalmer25 — APPROVED** [review]

**Pull request merged**